header-verifier/src/lib.rs

```diff
@@ -67,8 +67,14 @@ pub trait Headerverifier: bls_signature::BlsSignatureModule {
             "Only ESDT Safe contract can call this endpoint"
         );
 
-        self.pending_hashes(hash_of_hashes)
+        let was_hash_removed = self
```

Comment:
i mean, if that point is reached isn't it better to have an actual error message rather than the ugly "bad storage array" one, it will help the user more